diff --git i/test/client/dorar_client_use_test.dart w/test/client/dorar_client_use_test.dart
index 4ae8ffd..678f265 100644
--- i/test/client/dorar_client_use_test.dart
+++ w/test/client/dorar_client_use_test.dart
@@ -1,7 +1,11 @@
+import 'package:drift/drift.dart';
+import 'package:drift/native.dart';
 import 'package:dorar_hadith/dorar_hadith.dart';
 import 'package:test/test.dart';
 
 void main() {
+  setUpAll(() => CacheDatabase.configureConnection(() => DatabaseConnection(NativeDatabase.memory())));
+  tearDownAll(CacheDatabase.resetConnection);
   group('DorarClient.use', () {
     test('runs the provided callback and disposes client', () async {
       final count = await DorarClient.use((c) async {
